### ISSUE TITLE
How does advanced spmv support $y = \alpha A x + \beta y$

### ISSUE THREAD

**Issue by yhmtsai** [SparseBLAS/spblas-reference#4]
Does the current interface support advanced spmv like ($y = \alpha A x + \beta y$)?

AFAIK, it supported scaled like `multiply(A, scaled(alpha, x), y)` for $y = \alpha A x$ from https://github.com/SparseBLAS/spblas-example-project/blob/8ba01d8364171f3c888df31f4e95874c433a43db/src/simple_spmv.cpp#L18-L19
How does it apply to advanced spmv? something like `multiply(A, scaled(alpha, x), scaled(beta, y))` or another function?
`scaled(beta, y)` reads the $beta * y$ not $y$. When putting the value back to the $scaled(beta, y)$, does y store value/beta?
If the scaled version also works on output, it also makes the original `multiply(A, scaled(alpha, x), y)` like $y = \alpha A x + y$
Another function for this advance spmv make more sense to me currently.

**Comment by BenBrock** [comment]
Sorry for being late getting back to you.  I think the way we want to support this is with the four-parameter version of `multiply`.

```cpp
// C = AB + E
multiply(A, B, E, C);
```

```cpp
// three-parameter version is then essentially an alias to multiply(A, B, C, C)
// C = AB + C
multiply(A, B, C);
```

To set `beta = 0`, you'd then pass in `scaled(0, C)` for E.

```cpp
// C = AB + 0*C
multiply(A, B, scaled(0, C), C);
```

Let me know if this is a priority for the examples and I can work on implementing the bits you need here.  You can currently get a similar effect with

```cpp
scale(0, C);
multiply(A, B, C);
```

**Comment by yhmtsai** [comment]
Yes, I currently use something similar to the last part to ensure C = AB
I feel that `multiply(A, B, C)` giving `C=AB+C` a bit weird from multiply meaning.

**Comment by BenBrock** [comment]
Fair point, let's discuss the desired behavior of `multiply(A, B, C)`.  (I'm now seeing that std::linalg's `matrix_product` does `C = AB`, so I think you're probably right.)
